### DISCUSSION TITLE
Custom resource definitions not deleted

### DISCUSSION THREAD

**Discussion by nneram** [argoproj/argo-cd#6069] [Q&A]
** Custom resource definistions are not deleted with the application

Hello everyone, I would like to share my problem with my argoCD application hoping to find some help. It's all operational but I'm stuck when it comes time to delete the application. Indeed, the CRDs are not deleted with this one. It annoys me in my use and I would like to know if it was a normal behavior and if there is a way to remove it this CRDs.
If someone could help me about this I would be pleased !

**Comment by mel1nn** [comment]
Same problem here.

**Reply by alexkunde** [reply]
Hi, First off make sure prune on autosync is turned on or you click it when manually deleting

What we experienced often is finalizers on the custom resource not being met and thus the object is just waiting for that to happen instead of being deleted. A quick fix would be to patch finalizers on your object prio to deleting it.

**Reply by nneram** [reply]
I don't think it's a question of finalizer or deleting parameters on ArgoCD. I found this issue https://github.com/argoproj/argo-cd/issues/167 which deals with CDR deletion as a bug. And yes after looking deeper I found this: 
https://github.com/argoproj/argo-cd/blob/a125794dbaaeaf4d08ba71f6d719be54c49da2f5/controller/appcontroller.go#L778
https://github.com/argoproj/argo-cd/blob/a125794dbaaeaf4d08ba71f6d719be54c49da2f5/controller/appcontroller.go#L779

The code said if the resource is a CRD then it will never be deleted, so this is why I start this discussion to understand this behavior and maybe start an issue about it.

**Reply by alexkunde** [reply]
Ah sorry i thought we are talking custom resource not CRD

**Reply by wtam2018** [reply]
ArgoCD Application CRD does not get deleted AFAIK..   However, it is deleted when you uninstall Argo CD for example when you do `kubectl delete -f install.yaml`.
